app/src/test/java/com/google/android/ground/ui/offlineareas/selector/DownloadProgressDialogTest.kt

```diff
@@ -81,6 +92,26 @@ class DownloadProgressDialogTest : BaseHiltTest() {
     assertTrue(isDismissed)
   }
 
+  @Test
+  fun `stopDownloading cancels active download and updates UI state`() = runWithTestDispatcher {
+    val progressFlow = MutableStateFlow<Pair<Int, Int>>(Pair(0, 0))
+    whenever(offlineAreaRepository.downloadTiles(Bounds(-10.0, -20.0, 10.0, 20.0)))
+      .thenReturn(progressFlow)
+
+    viewModel.onDownloadClick()
+    advanceUntilIdle()
+
+    progressFlow.emit(Pair(50, 100))
+    assertEquals(0.5f, viewModel.downloadProgress.value)
+
+    viewModel.stopDownloading()
+
+    progressFlow.emit(Pair(75, 100))
+
+    assertNotEquals(0.75f, viewModel.downloadProgress.value)
+    assertFalse(viewModel.isDownloadProgressVisible.value!!)
```

Comment:
We should also check that the download coroutine is stopped as well

Reply:
downloadJob?

Reply:
Yeah